pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/ast/AbstractApexNodeBase.java

```diff
@@ -38,13 +38,12 @@ public AbstractApexNodeBase(Class<?> klass) {
      * Accept the visitor. *
      */
     public Object childrenAccept(ApexParserVisitor visitor, Object data) {
-        if (children != null) {
-            for (int i = 0; i < children.length; ++i) {
-                // we know that the children here are all ApexNodes
-                AbstractApexNodeBase apexNode = (AbstractApexNodeBase) children[i];
-                apexNode.jjtAccept(visitor, data);
-            }
+        for (int i = 0; i < children.length; ++i) {
```

Comment:
While this change probably makes sense (assuming that children is never null), it is not related to this PR, which is about Java and not Apex... It would be better to have a separate PR for this change and look into all other languages.
It seems, you changed it for all other languages, but it's hard to find this change in the 224 files...